internal/velero/storagelocation.go

```diff
@@ -32,21 +31,49 @@ import (
 	"github.com/vmware-tanzu/velero/pkg/plugin/clientmgmt"
 )
 
-// StorageLocation holds information for connecting with storage
-// for a backup storage location.
-type StorageLocation struct {
-	Client client.Client
-	Ctx    context.Context
-
+// DefaultBackupLocationInfo holds server default backup storage location information
+type DefaultBackupLocationInfo struct {
 	DefaultStorageLocation          string
 	DefaultStoreValidationFrequency time.Duration
+}
 
+// BackupStoreManager is a utility for connecting with storage
+// for a given backup storage location.
+type BackupStoreManager struct {
 	// use variables to refer to these functions so they can be
 	// replaced with fakes for testing.
 	NewPluginManager func(logrus.FieldLogger) clientmgmt.Manager
 	NewBackupStore   func(*velerov1api.BackupStorageLocation, persistence.ObjectStoreGetter, logrus.FieldLogger) (persistence.BackupStore, error)
 }
 
+// LocationStore contains data for a backup storage location
+// and its associated store, as well as default backup location values.
+type LocationStore struct {
+	location            *velerov1api.BackupStorageLocation
+	backupStore         persistence.BackupStore
+	defaultLocationInfo DefaultBackupLocationInfo
```

Comment:
This information will be duplicated in every BSL created. This will require updating every BSL should the `default` BSL change. I think this field can be dropped.

Reply:
The values for the two fields in this struct are not tied up to any BSL, it's a server level default. I made some changes but kept the `DefaultBackupLocationInfo` and added documentation. When you look at it again, please let me know if you still see any potential problem.